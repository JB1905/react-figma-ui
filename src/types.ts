import { colorNames, iconNames, sizes, tints, weights } from './constants';

export type ColorName = (typeof colorNames)[number];
export type IconName = (typeof iconNames)[number];
export type Size = (typeof sizes)[number];
export type Tint = (typeof tints)[number];
export type Weight = (typeof weights)[number];
