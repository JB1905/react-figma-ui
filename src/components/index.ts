export * from './Button';
export * from './Checkbox';
export * from './Disclosure';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Label';
export * from './Onboarding';
export * from './Radio';
export * from './SectionTitle';
export * from './SelectMenu';
export * from './Switch';
export * from './Textarea';
export * from './Type';
