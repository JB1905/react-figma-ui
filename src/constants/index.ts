export * from './colorNames';
export * from './iconNames';
export * from './sizes';
export * from './tints';
export * from './weights';
