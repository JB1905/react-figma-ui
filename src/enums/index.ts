export { Size } from './Size';
export { Tint } from './Tint';
export { Weight } from './Weight';
