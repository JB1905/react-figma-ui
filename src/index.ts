export { Button } from './components/Button';
export { Checkbox } from './components/Checkbox';
export { Disclosure } from './components/Disclosure';
export { Icon } from './components/Icon';
export { IconButton } from './components/IconButton';
export { Input } from './components/Input';
export { Label } from './components/Label';
export { Onboarding } from './components/Onboarding';
export { Radio } from './components/Radio';
export { SectionTitle } from './components/SectionTitle';
export { SelectMenu } from './components/SelectMenu';
export { Switch } from './components/Switch';
export { Textarea } from './components/Textarea';
export { Type } from './components/Type';
