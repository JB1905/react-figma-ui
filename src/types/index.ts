import { colorNames } from '../constants/colorNames';
import { iconNames } from '../constants/iconNames';
import { sizes } from '../constants/sizes';
import { tints } from '../constants/tints';
import { weights } from '../constants/weights';

export type ColorName = typeof colorNames;
export type IconName = typeof iconNames;
export type Size = typeof sizes;
export type Tint = typeof tints;
export type Weight = typeof weights;
