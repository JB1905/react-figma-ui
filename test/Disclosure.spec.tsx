import React from 'react';
import { render } from '@testing-library/react';

import { Disclosure, DisclosureItem } from '../src';

describe('Disclosure', () => {
  it('should render component matching snapshot', () => {
    const { container } = render(<Disclosure>{}</Disclosure>);

    expect(container).toMatchSnapshot();
  });
});
