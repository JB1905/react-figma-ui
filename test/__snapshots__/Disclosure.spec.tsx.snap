// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclosure should render Disclosure 1`] = `
<div>
  <ul
    class="disclosure"
  >
    <li
      class="disclosure__item disclosure--expanded"
    >
      <div
        class="disclosure__label disclosure--section"
      >
        <p>
          Heading 1
        </p>
      </div>
      <div
        class="disclosure__content"
      >
        <p>
          Content 1
        </p>
      </div>
    </li>
    <li
      class="disclosure__item"
    >
      <div
        class="disclosure__label"
      >
        <p>
          Heading 2
        </p>
      </div>
      <div
        class="disclosure__content"
      >
        <p>
          Content 2
        </p>
      </div>
    </li>
    <li
      class="disclosure__item"
    >
      <div
        class="disclosure__label disclosure--section"
      >
        <p>
          Heading 3
        </p>
      </div>
      <div
        class="disclosure__content"
      >
        <p>
          Content 3
        </p>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Disclosure should render Disclosure with custom className values 1`] = `
<div>
  <ul
    class="disclosure custom-class-name"
  >
    <li
      class="disclosure__item custom-class-name disclosure--expanded"
    >
      <div
        class="disclosure__label custom-class-name"
      >
        <p>
          Heading 1
        </p>
      </div>
      <div
        class="disclosure__content custom-class-name"
      >
        <p>
          Content 1
        </p>
      </div>
    </li>
    <li
      class="disclosure__item custom-class-name"
    >
      <div
        class="disclosure__label custom-class-name disclosure--section"
      >
        <p>
          Heading 2
        </p>
      </div>
      <div
        class="disclosure__content custom-class-name"
      >
        <p>
          Content 2
        </p>
      </div>
    </li>
    <li
      class="disclosure__item custom-class-name"
    >
      <div
        class="disclosure__label custom-class-name"
      >
        <p>
          Heading 3
        </p>
      </div>
      <div
        class="disclosure__content custom-class-name"
      >
        <p>
          Content 3
        </p>
      </div>
    </li>
  </ul>
</div>
`;
